File: java/CryptoMisuseVulnerable.java
import java.security.*;
import javax.crypto.*;

public class CryptoMisuseVulnerable {

    // Matches: java-md5-digest
    public MessageDigest vulnerableMd5() throws Exception {
        return MessageDigest.getInstance("MD5");
    }

    // Matches: java-sha1-digest
    public MessageDigest vulnerableSha1() throws Exception {
        return MessageDigest.getInstance("SHA-1");
    }

    // Matches: java-des-cipher
    public Cipher vulnerableDes() throws Exception {
        return Cipher.getInstance("DES");
    }

    // Matches: java-ecb-mode (pattern-regex)
    public Cipher vulnerableEcb() throws Exception {
        return Cipher.getInstance("AES/ECB/PKCS5Padding");
    }

    // Matches: java-securerandom-seed
    public SecureRandom vulnerableSeed() {
        byte[] seed = {1, 2, 3, 4};
        return new SecureRandom(seed);
    }

    // Safe: AES-GCM with proper random
    public Cipher safeCipher() throws Exception {
        return Cipher.getInstance("AES/GCM/NoPadding");
    }
}

File: go/crypto_misuse_vulnerable.go
package main

import (
	"crypto/des"
	"crypto/md5"
	"crypto/sha1"
	"math/rand"
)

// Matches: go-md5-usage
func vulnerableMd5(data []byte) []byte {
	h := md5.New()
	h.Write(data)
	return h.Sum(nil)
}

// Matches: go-sha1-usage
func vulnerableSha1(data []byte) []byte {
	h := sha1.New()
	h.Write(data)
	return h.Sum(nil)
}

// Matches: go-math-rand
func vulnerableRand() int {
	return rand.Intn(1000000)
}

// Matches: go-des-cipher
func vulnerableDes(key []byte) (interface{}, error) {
	return des.NewCipher(key)
}

// Safe: use crypto/rand and SHA-256
func safeRandom() ([]byte, error) {
	b := make([]byte, 32)
	_, err := crypto_rand.Read(b)
	return b, err
}
